cli/build.go

```diff
@@ -199,31 +199,30 @@ func (cmd BuildFlags) getTargetImageName() (image.Name, error) {
 	), nil
 }
 
-func (cmd BuildFlags) getBuildPlan(
-	contextDir string, imageName image.Name) (*builder.BuildPlan, error) {
-
-	// Remove image manifest if it already exists.
-	if err := cmd.cleanManifest(imageName); err != nil {
-		return nil, fmt.Errorf("failed to clean manifest: %v", err)
-	}
-
-	// Read in and parse dockerfile.
+func (cmd BuildFlags) createBuildPlan(contextDir string, imageName image.Name) (*builder.BuildPlan, error) {
+	// Create BuildContext.
 	contextDir, err := filepath.Abs(contextDir)
 	if err != nil {
 		return nil, fmt.Errorf("failed to resolve context dir: %s", err)
 	}
-	dockerfile, err := cmd.getDockerfile(contextDir)
-	if err != nil {
-		return nil, fmt.Errorf("failed to get dockerfile: %v", err)
-	}
-
-	// Create BuildContext.
 	buildContext, err := context.NewBuildContext("/", contextDir, cmd.imageStore)
 	if err != nil {
 		return nil, fmt.Errorf("failed to create initial build context: %s", err)
 	}
 	defer buildContext.Cleanup()
 
+	// Optionally remove everything before and after build.
+	if cmd.AllowModifyFS {
+		buildContext.MemFS.Remove()
+		defer buildContext.MemFS.Remove()
```

Comment:
you don't want to defer here, that will happen before `createBuildPlan` returns.